pygfx/helpers/_selection.py

```diff
@@ -0,0 +1,457 @@
+                    ),
+                    indices=np.array([[0, 1, 2, 3]], dtype=np.int32),
+                ),
+                gfx.MeshBasicMaterial(color=self._fill_color),
+            )
+            self._fill.material.opacity = self._fill_opacity
+            self.add(self._fill)
+        else:
+            self._fill = None
+
+        # Generate outline - this is a line where the first and the last point are
+        # at the same position.
+        if self._edge_color:
+            # Translate line style to dash pattern
+            dash_pattern = {
+                "solid": (),
+                "dashed": (5, 2),
+                "dotted": (1, 2),
+                "dashdot": (5, 2, 1, 2),
+            }.get(self._line_style, self._line_style)
+
+            self._outline = gfx.Line(
+                gfx.Geometry(
+                    positions=np.array(
+                        [(0, 0, 0), (1, 0, 0), (1, 1, 0), (0, 1, 0), (0, 0, 0)],
+                        dtype=np.float32,
+                    )
+                ),
+                gfx.LineMaterial(
+                    dash_pattern=dash_pattern,
+                    thickness=self._line_width,
+                    color=self._edge_color,
+                ),
+            )
+            self._outline.material.opacity = self._outline_opacity
+            self.add(self._outline)
+        else:
+            self._outline = None
+
+        # Generate info text
+        if self._show_info:
+            self._show_info = (
+                gfx.Text(
+                    gfx.TextGeometry(
+                        markdown="",
+                        font_size=self._info_font_size,
+                        anchor="bottomright",
+                        screen_space=True,
+                    ),
+                    gfx.TextMaterial(
+                        color=self._edge_color if self._edge_color else self._fill_color
+                    ),
+                ),
+                gfx.Text(
+                    gfx.TextGeometry(
+                        markdown="",
+                        font_size=self._info_font_size,
+                        anchor="topleft",
+                        screen_space=True,
+                    ),
+                    gfx.TextMaterial(
+                        color=self._edge_color if self._edge_color else self._fill_color
+                    ),
+                ),
+            )
+            self.add(self._show_info[0])
+            self.add(self._show_info[1])
+
+    def update_gizmo(self, event):
+        """Update the NDC transform."""
+        if event.type != "before_render":
+            return
+
+        if self._viewport and self._camera and self._active:
+            self._update_ndc_screen_transform()
+
+    def _update_ndc_screen_transform(self):
+        # Note: screen origin is at top left corner of NDC with Y-axis pointing down
+        x_dim, y_dim = self._viewport.logical_size
+        screen_space = AffineTransform()
```

Comment:
I did not realize this exists, I have been doing it manually with `pylinalg` all this while :joy:

Reply:
nevermind I thought this does the vector transform